src/query/interpreter.cpp

```diff
@@ -1871,6 +1879,40 @@ PreparedQuery PrepareIsolationLevelQuery(ParsedQuery parsed_query, bool in_expli
       RWType::NONE};
 }
 
+PreparedQuery PrepareStorageModeQuery(ParsedQuery parsed_query, const bool in_explicit_transaction,
+                                      InterpreterContext *interpreter_context) {
+  if (in_explicit_transaction) {
+    throw StorageModeModificationInMulticommandTxException();
+  }
+
+  auto *storage_mode_query = utils::Downcast<StorageModeQuery>(parsed_query.query);
+  MG_ASSERT(storage_mode_query);
+
+  const auto storage_mode = ToStorageStorageMode(storage_mode_query->storage_mode_);
+
+  auto callback = [storage_mode, interpreter_context]() -> std::function<void()> {
+    auto exists_active_transaction = interpreter_context->interpreters.WithLock([](const auto &interpreters_) {
+      return std::any_of(interpreters_.begin(), interpreters_.end(), [](const auto &interpreter) {
+        return interpreter->transaction_status_.load() == TransactionStatus::ACTIVE;
```

Comment:
It seems to me this is the wrong check: By discarding only interpreters that are active you are allowing changing storage mode when interpreter is in the started commit state, started rollback or verifying state and I think this can fail in many ways.

Reply:
Yes, you are right about discarding other interpreters. It should be that there are no active transactions. I first put that, but then there was bug on transactions, and didn't change as I forgot.

Reply:
Global storage mode is inside storage, same way as isolation level global is in storage. But why would then mvcc need to know about storage mode or anything like that, it is not interacting with Storage object, but with transaction, right?

Reply:
> this assumes that the prior transacaction has finished because otherwise which storage mode are we actually in? 

yes, that is why we want to all transactions to be finished.

Reply:
Good point, I think mvcc shouldn't know also about specific details of the storage mode but since you wrote in one other comment that you figured out this, I assume you found a way to refactor this?

Reply:
I thought about it even more, and I again I am not sure, because then really the problem is back again that mvcc needs to get reference somehow to storage, and that is problem because it shouldn't have that reference. I would maybe leave it as it is. 

I can check if there is some workaround, but when you think about it, transactions are the ones carrying information about what should transactions see, and it shouldn't be that mvcc is also looking at storage.